src/utils/recipient.js

```diff
@@ -273,6 +275,29 @@ export const pmUnreadsKeyFromMessage = (message: Message, ownUserId?: number): s
   }
 };
 
+/**
+ * The key for a PM thread in "unreads" data, given the key we use elsewhere.
+ *
+ * This produces the same key string that `pmUnreadsKeyFromMessage` would
+ * give, given the list of users that `pmKeyRecipientsFromMessage` would
+ * give and which we use in most of our other data structures.
+ */
+// See comment on pmUnreadsKeyFromMessage for details on this form.
+export const pmUnreadsKeyFromPmKeyIds = (
+  userIds: $ReadOnlyArray<number>,
+  ownUserId: number,
+): string => {
+  if (userIds.length === 1) {
+    // A 1:1 PM.  Both forms include just one user: the other user if any,
+    //   and self for a self-1:1.
+    return userIds[0].toString();
```

Comment:
Are you thinking we'll eventually combine `state.unread.pms` and `state.unread.huddles` into one bucket with a shared key space, as part of dissolving unhelpful distinctions between group and 1:1 PMs?

Reply:
Yeah, I think that'd make sense. There's this comment already in `pmConversationsSelectors.js`:
```js
      unread:
        // This business of looking in one place and then the other is kind
        // of messy.  Fortunately it always works, because the key spaces
        // are disjoint: all `unreadHuddles` keys contain a comma, and all
        // `unreadPms` keys don't.
        /* $FlowFixMe: The keys of unreadPms are logically numbers, but because it's an object they
         end up converted to strings, so this access with string keys works.  We should probably use
         a Map for this and similar maps. */
        unreadPms[recipient.unreadsKey] || unreadHuddles[recipient.unreadsKey],
```

That one is looking in a couple of cached-selector results, not `state.unread.pms` and `state.unread.huddles` themselves, but the selectors correspond directly to those.